training_pipeline/pipeline/configs.py

```diff
@@ -42,6 +43,59 @@
 EXAMPLE_GEN_BEAM_ARGS = None
 TRANSFORM_BEAM_ARGS = None
 
+"""
+EVAL_CONFIGS is to configuration for the Evaluator component
+to define how it is going to evalua the model performance.
+
+tfma.ModelSpec
+    signature_name is one of the signature in the SavedModel from
+    Trainer component, and it will be used to make predictions on 
+    given data. preprocessing_function_names allows us to include
+    a set of transformation(preprocessing) signatures in the Saved
+    Model from Trainer component. 
+
+    label_key and prediction_key will be used to compare the ground
+    truth and prediction results.
+
+slicing_specs
+    we use the entire dataset to evaluate the model performance. If
+    you want to evaluate the model based on different slices of data
+    set, you should prepare TFRecords to have multiple features which
+    of each corresponds to each slices(or categories), then write the
+    slicing_specs options accordingly.
+"""
+EVAL_CONFIGS = tfma.EvalConfig(
+    model_specs=[
+        tfma.ModelSpec(
+            signature_name="from_examples",
+            preprocessing_function_names=["transform_features"],
+            label_key="label_xf",
+            prediction_key="label_xf",
+        )
+    ],
+    slicing_specs=[tfma.SlicingSpec()],
+    metrics_specs=[
+        tfma.MetricsSpec(
+            metrics=[
+                tfma.MetricConfig(
+                    class_name="SparseCategoricalAccuracy",
+                    threshold=tfma.MetricThreshold(
+                        value_threshold=tfma.GenericValueThreshold(
+                            lower_bound={"value": 0.55}
```

Comment:
Minimum accuracy for the Resolver component to derive it should graduate the current model to prod or not?

Reply:
yeah `tfma.GenericValueThreshold` is normally defined to set the minimum performance threshold. That means a model whose performance is better than this is going to be the first model, and it also means that it is only used when there is no model deployed in production yet.

Reply:
And which component tells the pipeline -- okay, I have a better model whose SparseCategoricalCrossentropy is better than the previously deployed model. `Resolver` I guess?

Reply:
Resolver is a kind of component to bring existing Artifact. So here it is used to look up previously trained best model